tests/Unit/PointwiseFunctions/AnalyticData/WaveEquation/Test_RegularSphericalWaveKerrSchild.cpp

```diff
@@ -0,0 +1,247 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <array>
+#include <memory>
+
+#include "DataStructures/DataBox/Prefixes.hpp"  // IWYU pragma: keep
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"  // IWYU pragma: keep
+#include "Evolution/Systems/ScalarWave/System.hpp"      // IWYU pragma: keep
+#include "Parallel/RegisterDerivedClassesWithCharm.hpp"
+#include "PointwiseFunctions/AnalyticData/WaveEquation/RegularSphericalWaveKerrSchild.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/MathFunctions/Gaussian.hpp"
+#include "PointwiseFunctions/MathFunctions/MathFunction.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+#include "tests/Unit/TestCreation.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+struct RegularSphWaveKerr {
+  using type = CurvedScalarWave::AnalyticData::RegularSphericalWaveKerrSchild;
+  static constexpr OptionString help{
+      "A regular spherical wave in Kerr spacetime"};
+};
+
+void test_construct_from_options() noexcept {
```

Comment:
Could you run the result of the construction and serialization through a full test? Also, let's avoid 1 and 0 for values as read in.

Reply:
done